build.rs

```diff
@@ -1,212 +1,159 @@
-use quote::ToTokens;
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote, TokenStreamExt};
 use std::env;
 use std::fs::File;
 use std::io::Write;
 use std::path::Path;
-
-mod types {
-    use indexmap::IndexMap;
-    use proc_macro2::TokenStream;
-    use quote::{quote, ToTokens, TokenStreamExt};
-    use serde::{Deserialize, Deserializer};
-
-    // Manual blacklist for now. See https://github.com/dtolnay/syn/issues/607#issuecomment-475905135.
-    fn has_spanned(ty: &str) -> bool {
-        match ty {
-            "DataStruct" | "DataEnum" | "DataUnion" => false,
-            "FnDecl" => false,
-            "QSelf" => false,
-            _ => true,
-        }
+use syn_codegen::{Data, Definitions, Node, Type};
+
+// Manual blacklist for now. See https://github.com/dtolnay/syn/issues/607#issuecomment-475905135.
+fn has_spanned(ty: &str) -> bool {
+    match ty {
+        "DataStruct" | "DataEnum" | "DataUnion" => false,
+        "FnDecl" => false,
+        "QSelf" => false,
+        _ => true,
     }
+}
 
-    #[derive(Debug, PartialEq, Eq, Hash, Deserialize)]
-    pub struct Ident(String);
-
-    impl ToTokens for Ident {
-        fn to_tokens(&self, tokens: &mut TokenStream) {
-            proc_macro2::Ident::new(&self.0, proc_macro2::Span::call_site()).to_tokens(tokens)
-        }
+fn definition_tokens(definitions: &Definitions, tokens: &mut TokenStream) {
+    for node in &definitions.types {
+        node_tokens(node, tokens);
     }
-
-    #[derive(Debug, PartialEq, Deserialize)]
-    pub struct Definitions {
-        pub types: Vec<Node>,
-        pub tokens: IndexMap<Ident, String>,
+    for syn_token in definitions.tokens.keys() {
+        let ident = format_ident!("{}", syn_token);
+        tokens.append_all(quote! {
+            impl ToJS for syn::token::#ident {
+                fn to_js(&self) -> JsValue {
+                    js!(#ident {
+                        span: self.span()
+                    })
+                }
+            }
+        });
     }
+}
 
-    impl ToTokens for Definitions {
-        fn to_tokens(&self, tokens: &mut TokenStream) {
-            tokens.append_all(&self.types);
-            for key in self.tokens.keys() {
-                tokens.append_all(quote! {
-                    impl ToJS for syn::token::#key {
-                        fn to_js(&self) -> JsValue {
-                            js!(#key {
-                                span: self.span()
-                            })
-                        }
-                    }
-                });
+fn node_tokens(node: &Node, tokens: &mut TokenStream) {
+    let ident = format_ident!("{}", node.ident);
+
+    let data = match &node.data {
+        Data::Private => {
+            if ident == "LitStr"
```

Comment:
Nit: let's `match node.ident.as_str() { ... }` instead of a series of branches and `||` comparisons.